### ISSUE TITLE
Hang on boot v2.5-5098

### ISSUE THREAD

**Issue by beefstew** [CloverHackyColor/CloverBootloader#39]
I upgraded Clover from maybe 4722 (am unsure of specific version) to 5098 and now the machine won't boot. The computer's macOS version is 10.14.x (not 10.15). Clover menu appears, I select hard disk boot entry (or let it select based on timeout), bunch of EFI-related boot logs display, and process hangs after a row of +++++++++

I assume booting doesn't work off hard disk because I have old EFI drivers and kexts in place that are incompatible with the new 5098 version. I had overwritten my working rescue USB thumb drive, so I built a new one using Mojave 10.14 installer following the [vanilla desktop guide](https://hackintosh.gitbook.io/-r-hackintosh-vanilla-desktop-guide/building-the-usb-installer). I am using the exact same thumb drive that I used for initial installation a few years ago; drive appears to work just fine. Booting from USB drive also results in errors or hang.

![Screen Shot 2019-11-18 at 11 33 44 PM](https://user-images.githubusercontent.com/2262454/69117139-ce24ab00-0a5c-11ea-93ce-7f978421b5e4.png)

You can see in this image that, indeed, EFI/Clover are using the intended thumb drive device. Unfortunately, I can't ever seem to get a debug log created in `EFI/CLOVER/misc`.

Computer hardware is a Gigabyte [GB-BXi3-5010](https://www.gigabyte.com/Mini-PcBarebone/GB-BXi3-5010-rev-10#ov). Even though BIOS settings were working, I updated settings to mirror [recommendations for Haswell](https://hackintosh.gitbook.io/-r-hackintosh-vanilla-desktop-guide/config.plist-per-hardware/haswell) in vanilla guide.

Any suggestions?

Configuration for latest attempt: [CLOVER.zip](https://github.com/CloverHackyColor/CloverBootloader/files/3861941/CLOVER.zip)

**Comment by SergeySlice** [comment]
Anyway debug log must be.

**Comment by beefstew** [comment]
Earlier, I was not actually botting from the USB device. I have been re-performing testing with alternate configurations now that I'm booting from the correct device.

Debug log shows what looks to be a functioning Clover. Am still experiencing  the hang after End RandomSeed ++++

[debug.log](https://github.com/CloverHackyColor/CloverBootloader/files/3875380/debug.log)

I searched Clover source code, trying to determine what exactly this step is doing (since log terminates prior to this). I didn't find an answer (nor any documentation of the boot process), but think that this is the point where the bootloader has loaded the kernel and its extensions into memory hands over control. Because the hang appears to be instantaneous, I'm suspecting a memory address jump failure.

**Comment by SergeySlice** [comment]
You may find the line "++++" in Clover sources.
After that Clover does DSDT, kext and kernel patches and jump to kernel.
For my experience there is no reason for wrong memory address to jump.
You may switch on kext patching debug and see the process on the screen then decided if the process hang or kernel hang.

**Comment by SergeySlice** [comment]
I see you are using CsmVideoDxe.efi. Try to boot without it. It is poorly working with Intel Graphics.

**Comment by SergeySlice** [comment]
This line is not good

> 16:734  0:022  Board-ID set from config as Mac-CENSORED

:)

**Comment by beefstew** [comment]
Thanks for your help, SergeySlice.

1) I manually censored that Board-ID line prior to upload -- it's a valid value in the real file
2) I removed the CsmVideoDxe.efi driver
3) Upgraded to 5100

Between 2 and 3, the machine finally boots. Still no sound (original reason for upgrade), but I have a functioning computer again.

[CLOVER-working.zip](https://github.com/CloverHackyColor/CloverBootloader/files/3958923/CLOVER-working.zip)
